### ISSUE TITLE
How to serve on 0.0.0.0?

### ISSUE THREAD

**Issue by xgdgsc** [GenieFramework/Stipple.jl#2]

**Comment by essenciary** [comment]
@xgdgsc Stipple runs on top of Genie so it inherits all the configuration logic. In this case add `Genie.config.server_host = "0.0.0.0"` somewhere before starting the server. 
https://genieframework.github.io/Genie.jl/dev/API/configuration.html#Genie.Configuration.Settings

**Comment by xgdgsc** [comment]
Thanks. I have another issue when I update a dataframe on each call to get route "/", the table should update,   in REPL I tested it works. But when I run it in a julia script using `julia -i serve_df.jl` . Although the same get route "/" is called properly (I can get dataframe size increased as it query from database and print out). But the table on page isn' t updated to the new dataframe size unlike in REPL testing it works.

**Comment by essenciary** [comment]
Not sure about the details of your implementation but Stipple/Genie works with Revise - make sure you use it: https://github.com/timholy/Revise.jl

**Comment by xgdgsc** [comment]
![image](https://user-images.githubusercontent.com/1189869/89392943-bac98500-d73c-11ea-81ea-c27e78083166.png)
I' ve put `using Revise` on top. I have this error in console.

**Comment by essenciary** [comment]
That's OK, it's a warning, haven't seen any actual problems caused by that yet.

**Comment by xgdgsc** [comment]
I feel the updating issue is because an unavailable websocket port. Now in dev env it works. But in prod env it throws errors:
```
julia> ┌ Error: 2020-08-06 12:18:51 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:52 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:167
┌ Error: 2020-08-06 12:18:52 error handling request
│   exception =
│    KeyError: key :__params not found
│    Stacktrace:
│     [1] getindex at ./iddict.jl:92 [inlined]
│     [2] task_local_storage at ./task.jl:204 [inlined]
│     [3] _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│     [4] response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│     [5] setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:173
│     [6] setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│     [7] #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│     [8] handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│     [9] handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│     [10] #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│     [11] macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│     [12] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
└ @ HTTP.Servers ~/.julia/packages/HTTP/atT5q/src/Servers.jl:373
┌ Error: 2020-08-06 12:18:52 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:54 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:167
┌ Error: 2020-08-06 12:18:54 error handling request
│   exception =
│    KeyError: key :__params not found
│    Stacktrace:
│     [1] getindex at ./iddict.jl:92 [inlined]
│     [2] task_local_storage at ./task.jl:204 [inlined]
│     [3] _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│     [4] response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│     [5] setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:173
│     [6] setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│     [7] #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│     [8] handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│     [9] handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│     [10] #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│     [11] macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│     [12] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
└ @ HTTP.Servers ~/.julia/packages/HTTP/atT5q/src/Servers.jl:373
┌ Error: 2020-08-06 12:18:55 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:55 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:167
┌ Error: 2020-08-06 12:18:55 error handling request
│   exception =
│    KeyError: key :__params not found
│    Stacktrace:
│     [1] getindex at ./iddict.jl:92 [inlined]
│     [2] task_local_storage at ./task.jl:204 [inlined]
│     [3] _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│     [4] response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│     [5] setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:173
│     [6] setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│     [7] #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│     [8] handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│     [9] handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│     [10] #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│     [11] macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│     [12] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
└ @ HTTP.Servers ~/.julia/packages/HTTP/atT5q/src/Servers.jl:373
┌ Error: 2020-08-06 12:18:55 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:55 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:167
┌ Error: 2020-08-06 12:18:55 error handling request
│   exception =
│    KeyError: key :__params not found
│    Stacktrace:
│     [1] getindex at ./iddict.jl:92 [inlined]
│     [2] task_local_storage at ./task.jl:204 [inlined]
│     [3] _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│     [4] response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│     [5] setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:173
│     [6] setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│     [7] #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│     [8] handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│     [9] handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│     [10] #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│     [11] macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│     [12] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
└ @ HTTP.Servers ~/.julia/packages/HTTP/atT5q/src/Servers.jl:373
┌ Error: 2020-08-06 12:18:55 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:55 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:167
┌ Error: 2020-08-06 12:18:55 error handling request
│   exception =
│    KeyError: key :__params not found
│    Stacktrace:
│     [1] getindex at ./iddict.jl:92 [inlined]
│     [2] task_local_storage at ./task.jl:204 [inlined]
│     [3] _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│     [4] response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│     [5] setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:173
│     [6] setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│     [7] #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│     [8] handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│     [9] handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│     [10] #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│     [11] macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│     [12] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
└ @ HTTP.Servers ~/.julia/packages/HTTP/atT5q/src/Servers.jl:373
┌ Error: 2020-08-06 12:18:55 KeyError(:__params)
└ @ Genie.AppServer ~/.julia/packages/Genie/rqHFz/src/AppServer.jl:142
┌ Error: 2020-08-06 12:18:55 KeyError: key :__params not found
│ getindex at ./iddict.jl:92 [inlined]
│ task_local_storage at ./task.jl:204 [inlined]
│ _params_ at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:110 [inlined]
│ response_mime() at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:1039
│ route_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/Router.jl:141
│ handle_request(::HTTP.Messages.Request, ::HTTP.Messages.Response, ::Sockets.IPv4) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:140
│ handle_request at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:133 [inlined]
│ #19 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/macros.jl:127 [inlined]
│ run_work_thunk(::Genie.AppServer.var"#19#20"{HTTP.Messages.Request,HTTP.Messages.Response}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/process_messages.jl:79
│ remotecall_fetch(::Function, ::Distributed.LocalProcess; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Distributed.LocalProcess) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:379
│ remotecall_fetch(::Function, ::Int64; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ remotecall_fetch(::Function, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421
│ setup_http_handler(::HTTP.Messages.Request, ::HTTP.Messages.Response) at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:155
│ setup_http_handler at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:154 [inlined]
│ #6 at /home/sgao/.julia/packages/Genie/rqHFz/src/AppServer.jl:76 [inlined]
│ handle at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:253 [inlined]
│ handle(::HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}, ::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}) at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:276
│ #4 at /home/sgao/.julia/packages/HTTP/atT5q/src/Handlers.jl:345 [inlined]
│ macro expansion at /home/sgao/.julia/packages/HTTP/atT5q/src/Servers.jl:367 [inlined]
│ (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Genie.AppServer.var"#6#12"}},HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at ./task.jl:356
│ 
```

**Comment by xgdgsc** [comment]
I ran it like `GENIE_ENV=prod JULIA_PROJECT=./ julia -i AlphaBack.jl  `

**Comment by xgdgsc** [comment]
I just copied `config/env/dev.jl` to `prod.jl` and it works. Thanks.
